tests/unit/modules/test_freezer.py

```diff
@@ -0,0 +1,239 @@
+        isdir.return_value = False
+        self.assertEqual(freezer.list_(), [])
+
+        # There is freezer directory, but is empty
+        isdir.return_value = True
+        listdir.return_value = []
+        self.assertEqual(freezer.list_(), [])
+
+        # There is freezer directory with states
+        isdir.return_value = True
+        listdir.return_value = [
+            'freezer-pkgs.yml', 'freezer-reps.yml',
+            'state-pkgs.yml', 'state-reps.yml',
+            'random-file'
+        ]
+        self.assertEqual(freezer.list_(), ['freezer', 'state'])
+
+    @patch('salt.utils.json.dump')
+    @patch('salt.modules.freezer.fopen')
+    @patch('salt.modules.freezer.isfrozen')
+    def test_freeze(self, isfrozen, fopen, dump):
+        '''
+        Test to freeze a current installation
+        '''
+
+        # Fails when creating the freeze cache directory
+        salt_mock = {
+            'cmd.retcode': MagicMock(return_value=1),
+        }
+        with patch.dict(freezer.__salt__, salt_mock):
+            self.assertRaises(CommandExecutionError, freezer.freeze)
+            salt_mock['cmd.retcode'].assert_called_once()
+
+        # Fails when there is already a frozen state
+        isfrozen.return_value = True
+        salt_mock = {
+            'cmd.retcode': MagicMock(return_value=0),
+        }
+        with patch.dict(freezer.__salt__, salt_mock):
+            self.assertRaises(CommandExecutionError, freezer.freeze)
+            salt_mock['cmd.retcode'].assert_called_once()
+
+        # Freeze the current new state
+        isfrozen.return_value = False
+        salt_mock = {
+            'cmd.retcode': MagicMock(return_value=0),
+            'pkg.list_pkgs': MagicMock(return_value={}),
+            'pkg.list_repos': MagicMock(return_value={}),
+        }
+        with patch.dict(freezer.__salt__, salt_mock):
+            self.assertTrue(freezer.freeze())
+            salt_mock['cmd.retcode'].assert_called_once()
+            salt_mock['pkg.list_pkgs'].assert_called_once()
+            salt_mock['pkg.list_repos'].assert_called_once()
+            fopen.assert_called()
+            dump.asster_called()
+
+        # Freeze the current old state
+        isfrozen.return_value = True
+        salt_mock = {
+            'cmd.retcode': MagicMock(return_value=0),
+            'pkg.list_pkgs': MagicMock(return_value={}),
+            'pkg.list_repos': MagicMock(return_value={}),
+        }
+        with patch.dict(freezer.__salt__, salt_mock):
+            self.assertTrue(freezer.freeze(force=True))
+            salt_mock['cmd.retcode'].assert_called_once()
+            salt_mock['pkg.list_pkgs'].assert_called_once()
+            salt_mock['pkg.list_repos'].assert_called_once()
+            fopen.assert_called()
+            dump.asster_called()
+
+    @patch('salt.utils.json.load')
+    @patch('salt.modules.freezer.fopen')
+    @patch('salt.modules.freezer.isfrozen')
+    def test_restore(self, isfrozen, fopen, load):
```

Comment:
This UT should be also broken into couple of those.

Reply:
done